src/rtkcmn.c

```diff
@@ -1152,15 +1173,15 @@ extern void matmul(const char *tr, int n, int k, int m, double alpha,
                    const double *A, const double *B, double beta, double *C)
 {
     double d;
-    int i,j,x,f=tr[0]=='N'?(tr[1]=='N'?1:2):(tr[1]=='N'?3:4);
-    
-    for (i=0;i<n;i++) for (j=0;j<k;j++) {
+    int i,j,x,f=(tr[0] != 'N') + (tr[1] != 'N') * 2;
+
+    for (j=0;j<k;j++) for (i=0;i<n;i++) {
         d=0.0;
         switch (f) {
-            case 1: for (x=0;x<m;x++) d+=A[i+x*n]*B[x+j*m]; break;
+            case 0: for (x=0;x<m;x++) d+=A[i+x*n]*B[x+j*m]; break;
+            case 1: for (x=0;x<m;x++) d+=A[x+i*m]*B[x+j*m]; break;
             case 2: for (x=0;x<m;x++) d+=A[i+x*n]*B[j+x*k]; break;
-            case 3: for (x=0;x<m;x++) d+=A[x+i*m]*B[x+j*m]; break;
-            case 4: for (x=0;x<m;x++) d+=A[x+i*m]*B[j+x*k]; break;
+            case 3: for (x=0;x<m;x++) d+=A[x+i*m]*B[j+x*k]; break;
```

Comment:
fwiw the changes to matmul() appear to be self contained, could be split out into a separate PR if that helped. The main change appears to be reordering the loop memory access, perhaps to optimize for some of the cases.

Reply:
Yep, that's exactly right. This pretty much just changes the memory access order. Which then provided around ~5% performance improvements in my test cases (using it for ppk with both forward and backward passes).
I opened #197 for these and the changes that remove the unnecessary temporary used in `rtkpos.c`. Have a look and let me know what you think!